CMakeLists.txt

```diff
@@ -0,0 +1,31 @@
+cmake_minimum_required(VERSION 3.1.3 FATAL_ERROR)
+project(hidapi LANGUAGES C)
+
+if(NOT DEFINED CMAKE_BUILD_TYPE OR CMAKE_BUILD_TYPE STREQUAL "")
+    set(CMAKE_BUILD_TYPE "Release" CACHE STRING "Debug;Release;MinSizeRel;RelWithDebInfo" FORCE)
+endif()
+set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
```

Comment:
What is the point of this? Why don't you leave it to CMake to set the default?

Reply:
> What is the point of this? 

It is for convenience of those, who are building hidapi as a standalone package/library. Try cmake-gui and you'll see.

> Why don't you leave it to CMake to set the default?

Because by default CMake doesn't set it at all. Maybe it does makes sence somtime, but I haven't seen a single real project where it would.

---
As for the convenience part: see my other comment https://github.com/libusb/hidapi/pull/220#discussion_r541672007

Reply:
I still don't get the point of overriding CMake's default behavior, but I guess it doesn't hurt? :shrug: It'll just be a bit of confusing code for people reading this in the future.

Reply:
> but I guess it doesn't hurt?

That's my point.

> CMake's default behavior

Humor me. Can you tell what's CMake's default behavior if you don't specify CMAKE_BUILD_TYPE at all?

Reply:
@Youw 
There are multi-config generators (like MSVC) that ignores `CMAKE_BUILD_TYPE`. The value of this variable is also not limited to the ones you've listed - for example I use also `TSan` and `ASan` build type for sanitizers. I agree this part of code isn't necessary here but I don't think it will break something on the other hand.

Reply:
> ignores `CMAKE_BUILD_TYPE`

The problem that it is not exactly ignored. It is still available in the CMake scripts, and what bothers me the most: CMake doesn't default it to empty value when MSVC generator is used, but rather to "Debug" (gcc + make - it is defaulted to empy value).
I find such behavior way more confusing, but I guess CMake's developers must have had their reasons for that.

> I use also `TSan` and `ASan` build type for sanitizers

I must admit - I haven't seen any custom CMAKE_BUILD_TYPE usage/implementation, so I guess that is a good point too.
Can you point to me a reference how your/typical usage look like?

As I mentioned earlier, a standalone build likely not to be used with a non-standard build types but I agree that prohibiting it wouldn't be a good thing, so there is [that](https://github.com/libusb/hidapi/commit/8ae8ffb2cf0ef6596baa55a9c8db3fc2dfc31e9f).